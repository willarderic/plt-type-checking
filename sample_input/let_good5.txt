(let* ( (s "kitty paws")
     )
  s
)
