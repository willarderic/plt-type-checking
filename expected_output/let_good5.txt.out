TOKEN: LPAREN    LEXEME: (
TOKEN: LETSTAR   LEXEME: let*

___Entering new scope...

TOKEN: LPAREN    LEXEME: (
ID_EXPR_LIST -> epsilon
TOKEN: LPAREN    LEXEME: (
TOKEN: IDENT     LEXEME: s
TOKEN: STRCONST  LEXEME: "kitty paws"
CONST -> STRCONST
EXPR -> CONST
TOKEN: RPAREN    LEXEME: )
ID_EXPR_LIST -> ID_EXPR_LIST ( IDENT EXPR )
___Adding s to symbol table
TOKEN: RPAREN    LEXEME: )
TOKEN: IDENT     LEXEME: s
EXPR -> IDENT
LET_EXPR -> let* ( ID_EXPR_LIST ) EXPR

___Exiting scope...

PARENTHESIZED_EXPR -> LET_EXPR
TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )
START -> EXPR

---- Completed parsing ----

